src/main/java/com/cse/helium/app/services/keyvault/KeyVaultService.java

```diff
@@ -44,27 +52,60 @@ public KeyVaultService(IEnvironmentReader environmentReader)
     logger.info("Auth type is " + this.authType);
 
     if (this.authType.equals(Constants.USE_MSI)) {
-      azureTokenCredentials = new MSICredentials(AzureEnvironment.AZURE);
+      credential = new DefaultAzureCredentialBuilder()
+          .excludeEnvironmentCredential()
+          .excludeAzureCliCredential()
+          .excludeSharedTokenCacheCredential()
+          .build();
     } else if (this.authType.equals(Constants.USE_CLI)) {
-      try {
-        azureTokenCredentials = AzureCliCredentials.create();
-      } catch (final IOException ex) {
-        logger.error(ex.getMessage());
-        throw ex;
-      }
+      credential = new DefaultAzureCredentialBuilder()
+        .excludeEnvironmentCredential()
+        .excludeManagedIdentityCredential()
+        .excludeSharedTokenCacheCredential()
+        .build();
     } else if (this.authType.equals(Constants.USE_MSI_APPSVC)) {
       try {
-        azureTokenCredentials = new AppServiceMSICredentials(AzureEnvironment.AZURE);
+        credential = new DefaultAzureCredentialBuilder()
+        .excludeEnvironmentCredential()
+        .excludeAzureCliCredential()
+        .excludeSharedTokenCacheCredential()
+        .build();
       } catch (final Exception ex) {
         logger.error(ex.getMessage());
         throw ex;
       }
     } else {
       this.authType = Constants.USE_MSI;
-      azureTokenCredentials = new MSICredentials(AzureEnvironment.AZURE);
+      credential = new DefaultAzureCredentialBuilder()
+      .excludeEnvironmentCredential()
+      .excludeAzureCliCredential()
+      .excludeSharedTokenCacheCredential()
+      .build();
     }
 
-    keyVaultClient = new KeyVaultClient(azureTokenCredentials);
+    //build secret clients
+    secretClient = new SecretClientBuilder()
+        .vaultUrl(getKeyVaultUri())
+        .credential(credential)
+        .buildClient();
+
+    secretAsyncClient = new SecretClientBuilder()
+        .vaultUrl(getKeyVaultUri())
+        .credential(credential)
+        .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
```

Comment:
Can we make the log level configurable based on the other log level parameter that we have "--log-level"? Otherwise, this will stay here permanently.

Reply:
Yes, it can be configurable.  No, this is not the same thing.  This is what information to include in the logging should the SDK encounter an error.  My preference is to leave it as written.  If we want to change it then we will can do that in another task as we will also need to define the mechanism/mapping between overall system settings and this setting.  The valid values are: 

BASIC | Logs only URLs, HTTP methods, and time to finish the request.
-- | --
BODY | Logs everything in BASIC, plus all the request and response body. Note that only payloads in plain text or plain text encoded in GZIP will be logged.
BODY_AND_HEADERS | Logs everything in HEADERS and BODY.
HEADERS | Logs everything in BASIC, plus all the request and response headers.
NONE | Logging is turned off.